mdata/aggmetric.go

```diff
@@ -356,7 +361,14 @@ func (a *AggMetric) persist(pos int) {
 	// create an array of chunks that need to be sent to the writeQueue.
 	pending := make([]*ChunkWriteRequest, 1)
 	// add the current chunk to the list of chunks to send to the writeQueue
-	cwr := NewChunkWriteRequest(a, a.key, chunk, a.ttl, a.chunkSpan, time.Now())
+	cwr := NewChunkWriteRequest(
+		a.SyncChunkSaveState(chunk.Series.T0, true),
+		a.key,
+		a.ttl,
+		chunk.Series.T0,
+		chunk.Encode(a.chunkSpan),
```

Comment:
As i have mentioned before on slack, i'm a bit concerned about the extra encoding overhead in the critical - synchronous - section, which used to be in the async queue processing
unless there is any news that i missed, i think you should do a quick test using the docker stack and fakemetrics to confirm the difference in behavior with this patch applied under a significant chunk saving workload.

Reply:
ok, i'll try that and compare if the ingest speed has changed due to that

Reply:
I have filled 4h of fakemetrics into my kafka topic in the QA cluster, that kafka topic keeps the data for 7.7h, so then i had 3.7h of time to replay the contents of the topic multiple times with different images and compare how long the replaying takes. I did all those test runs with write pods, just manually killing them and then watching how long it takes for them to replay the kafka backlog.

I did 3 runs with each of `v0.11.0-295-gdd9b92d` and the latest image from this branch `whisper-importer-test8 (v0.11.0-297-gbb4a504)`
Unfortunately there seems to be quite a lot of variability between the different runs, so I think they might be unreliable and maybe I should better try comparing that on my local laptop or on bastion to get a more isolated environment.

```
us.gcr.io/metrictank-gcr/metrictank:v0.11.0-295-gdd9b92d:                                      

test1: 21:25:40 - 21:36:40 (11:00)
test2: 21:38:30 - 21:52:10 (13:40)
test3: 21:54:30 - 22:04:50 (10:20)
                                                                                                         
tus.gcr.io/metrictank-gcr/metrictank:whisper-importer-test8 (v0.11.0-297-gbb4a504):             

test1: 22:06:30 - 22:19:50 (13:20)
test2: 22:22:00 - 22:32:00 (12:00)
test3: 22:33:10 - 22:43:40 (10:30)
```

Reply:
I've done another test on my local laptop, comparing the two branches. again, i filled kafka with a lot of data and made the different versions replay the data and insert it into cassandra. between each run i truncated all the cassandra tables. these are the results:

```
testing with latest master (v0.11.0-295-gdd9b92d)
run1: 13:56:54 - 14:02:42 (05:48)
run2: 14:11:33 - 14:17:18 (05:45)
run3: 15:06:49 - 15:12:55 (06:06)

testing with latest importer_bigtable (v0.11.0-298-gc2ba6e8)
run1: 14:33:00 - 14:38:43 (05:43)   
run2: 14:48:41 - 14:54:28 (05:47)   
run3: 14:56:04 - 15:02:11 (06:07)
```

Reply:
can you share snapshots that are zoomed in on the timeframes where it creates all the cwr's ? i'm curious about the impact of the more expensive cwr creation.

Reply:
I didn't capture snapshots, that means I'll need to do that again

Reply:
i'm also curious to see if there's any change in memory stats

Reply:
I doubt there would be any negative CPU impact to encoding chunks before adding to the write queue.

with the encoding happening before we add to the write queue, we are constrained to 8 (number of threads processing ingested metrics) concurrent goroutines encoding chunks.   With chunks being encoded after being added to the write queue, there are 10 (default) write threads that can encode chunks at the same time.

There could be more memory used as the amount of time that both the *chunk.Chunk and []byte slice exist is greater.  but i wouldnt expect a significant impact from this.

Reply:
I ran another benchmark. on my localhost i filled kafka with `3h` of data for `10000` metrics at `1sec` resolution. Then I started MT and made it write all that data to cassandra. Once I started the latest `master` and once this branch. Between each run I truncated all cassandra tables. 
The results are really very similar:

with importer_bigtable changes:
https://snapshot.raintank.io/dashboard/snapshot/IFx69Jh8yd1NGxy4ZBZCeCccNvW1iucS                                                        

with master
https://snapshot.raintank.io/dashboard/snapshot/jvNtJf0KoNkbYaNs3M1duCBTE5XKqFuM